source/exe/main.cc

```diff
@@ -33,10 +35,36 @@ class ProdComponentFactory : public ComponentFactory {
 
 } // Server
 
+namespace {
+
+int validateConfig(OptionsImpl& options, Server::ProdComponentFactory& component_factory,
+                   const LocalInfo::LocalInfoImpl& local_info) {
+  Server::ValidationHotRestart restarter;
+  Logger::Registry::initialize(options.logLevel(), restarter.logLock());
```

Comment:
The export of the log and access locks isn't part of the `HotRestart` interface. I _think_ you can both simplify here by just using locally defined lock objects in both cases and also removing locking from the `Server::ValidationHotRestart` implementation, can you take a look to see if this is the case?